sw/device/lib/testing/clkmgr_testutils.h

```diff
@@ -0,0 +1,47 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_TESTING_CLKMGR_TESTUTILS_H_
+#define OPENTITAN_SW_DEVICE_LIB_TESTING_CLKMGR_TESTUTILS_H_
+
+#include "sw/device/lib/dif/dif_base.h"
+#include "sw/device/lib/dif/dif_clkmgr.h"
+
+/**
+ * Checks a transactional block's clock status.
+ *
+ * @param clkmgr A clkmgr handle.
+ * @param params Hardware instance parameters.
+ * @param clock The transactional clock ID.
+ * @param exp_clock_enabled Actual clock gating status.
+ */
+void clkmgr_testutils_check_trans_clock_status(
+    dif_clkmgr_t *clkmgr, dif_clkmgr_params_t params,
+    dif_clkmgr_hintable_clock_t clock, bool exp_clock_enabled);
+
+/**
+ * Sets clock gating hint for a transactional block and verify clock status.
+ *
+ * For a transactional block, the clock is gated only if the software enables
+ * the clock gating AND the block is idle. This function sets the clock gating
+ * bit and after a random amount of time (microseconds), checks if the clock is
+ * really gated (or not).
```

Comment:
I think it's important that the range of delays could be set or modified in such a way that this function becomes flaky if it's run when a block isn't idle and the range of random values spans the time when the clock becomes idle. Could you add a line to the header expressing that the range of time shouldn't include the time when the transactional block completes its work and becomes idle.

Reply:
Note that this testutil can be called for the idle case too (check that in idle state, the clock can indeed be turned off). In that case, the delay range can be sufficiently wide.